### ISSUE TITLE
docs: how to pre install JetBrains editors into VM / container image so that it doesn't have to download every time

### ISSUE THREAD

**Issue by bpmct** [coder/coder#13207]
Installing a JetBrains IDE can be slow. I believe sharkymark has some v2 templates with IntellIJ pre installed. Can we formalize these docs?

**Comment by matifali** [comment]
@bpmct AFAIR, Those templates were using the now deprecated JetBrains projector. 
To document this will require investigating where the Gateway puts the downloaded IDE backend.

**Comment by matifali** [comment]
@roshvin This package is JetBrainsClient its being downloaded on the client machine and not the workspace. I am currently unable to reproduce this. For me this gets downloaded only the first time or if there is a newer version is available. Its being handled by Gateway automatically.

**Comment by code-asher** [comment]
Yeah both the client and backend should only be downloaded once, if not then there could be a bug but I just tried and have not been able to reproduce as of yet.

The backend can be pre-installed anywhere, you just have to register it (basically it just creates a symlink in the right spot).  For example, suppose you download and extract the editor to `~/.cache/editors/IU-232.10203.10`:

```
~/.cache/editors/IU-232.10203.10/bin/remote-dev-server.sh registerBackendLocationForGateway
```

https://www.jetbrains.com/help/idea/remote-development-a.html#register_backend_IDE

I am not sure where the client goes though.

**Comment by code-asher** [comment]
Also, if you ever want to check what is registered with Gateway without having to launch Gateway, I am not sure there is a better way but this does the trick for me (note the hash might be different for others):

```shell
$ ~/.cache/JetBrains/RemoteDev/remote-dev-worker/remote-dev-worker_fcf8a611ab8f9110082dc5a71cb4e287dc9ee35120fb7880e4e56e6b92abec56 installed-ides | jq '.data[]  | fromjson | .idePath'
"/home/coder/.cache/JetBrains/RemoteDev/dist/5bffcb8389003_PhpStorm-241.15989.66"
"/home/coder/.cache/JetBrains/RemoteDev/dist/CL-233.14475.31"
"/home/coder/.cache/JetBrains/RemoteDev/dist/CL-241.14494.229"
"/home/coder/.cache/JetBrains/RemoteDev/dist/GO-233.14808.20"
"/home/coder/.cache/JetBrains/RemoteDev/dist/IU-233.15026.9"
"/home/coder/.cache/JetBrains/RemoteDev/dist/IU-241.15989.69"
```

**Comment by matifali** [comment]
Some related docs from JetBrains. 
https://www.jetbrains.com/help/idea/faq-about-remote-development.html#register_backend

**Comment by matifali** [comment]
I don't see any. Adding it to my list. 
To summarize,
1. Document how to modify a template with JetBrains gateway backend pre downloaded. 
2. Register the backend as per this [document](https://www.jetbrains.com/help/idea/remote-development-a.html#register_backend_IDE).

**Comment by code-asher** [comment]
We talk about it a bit here: https://coder.com/docs/ides/gateway#using-an-existing-jetbrains-installation-in-the-workspace

But nothing about how to install it, only how to register it once it is already installed.

**Comment by matifali** [comment]
> We talk about it a bit here: [coder.com/docs/ides/gateway#using-an-existing-jetbrains-installation-in-the-workspace](https://coder.com/docs/ides/gateway#using-an-existing-jetbrains-installation-in-the-workspace)
@code-asher Thank you. I think the best place to document how to pre-download it can go along with that, too. Do you agree @bpmct?

**Comment by matifali** [comment]
Doing this in https://github.com/coder/jetbrains-coder/issues/504.

**Comment by matifali** [comment]
While investiagting this. Looks like we already documented this in a form at [coder.com/docs/user-guides/workspace-access/jetbrains#jetbrains-gateway-in-an-offline-environment](https://coder.com/docs/user-guides/workspace-access/jetbrains#jetbrains-gateway-in-an-offline-environment)

We can repurpose those docs to also serve non-offline deployments.

Can you help here @EdwardAngert?
